doc/ERROR_CONCEPT.md

```diff
@@ -0,0 +1,250 @@
+2. Some errors might be duplicated just because they belong to different plugins (or the dev neglected to lookup a similar in the specification) 
+(#29 + #57 + #72 + #169, #10 + #61, etc.)
+3. Many errors are very similar or could be summarized and are differentiated via the reason field 
+(#54 + #55, #31 - #33, #93 + #7 + #17, #74 + #75, #77 + #10 + #61, etc.)
+4. Some errors are marked as unused (#22 + #48 + #58 + #66 + #67 + #68)
+5. In many cases the `Description` is not very relevant at all (#199, #196, #187, etc.)
+
+## Evaluation
+
+From now on I will take the following error message as example:
+
+```
+1. The command kdb set failed while accessing the key database with the info:
+2. Sorry, the error (#121) occurred ;(
+3. Description: validation failed
+4. Reason: Validation of key "<key>" with string "<value>" failed.
+5. Ingroup: plugin
+6. Module: enum
+7. At: ....../src/plugins/enum/enum.c:218
+8. Mountpoint: <parentKey>
+9. Configfile: ...../<file>.25676:1549919217.284067.tmp
+```
+
+The lines are numbered and will be referred in the following sections.
+
+Generally I would suggest that this error message is way too much information for a user. We can agree on the fact 
+that #4 is by far the most relevant line.
+
+### Line 9: Configfile
+
+Reasons to keep it:
+* If users want to see where the concrete file is located 
+
+Reasons against it:
+* Very verbose line
+* I highly doubt that most common users will require this information (unnecessary)
+
+Suggestions:
+Remove it completely. If users want to know the location they still can use `kdb file` on the concrete setting.
+
+### Line 8: Mountpoint
+
+Reasons to keep it:
+* If users want to see the parent key of their configuration
+* If I remember correctly: all error information is saved to this key, so users could investigate further error
+information which is saved to this key (in the metadata)
+
+Reasons against it:
+* Another extra line in the error message.
+* I highly doubt that most common users will require this information (unnecessary)
+
+Suggestions:
+Remove it for standard error messages. Maybe a `kdb set -v` could include such information for developers, users.
+Alternatively there could be a command for users to see the mountpoint, eg. `kdb get-mountpoint <key>`
+
+### Line 7: At
+
+Reasons to keep it:
+* Debug purposes
+
+Reasons against it:
+* Another extra line in the error message.
+* I highly doubt that most common users will require this information (unnecessary)
+
+Suggestions:
+Remove it for standard error messages. Maybe a `kdb set -v` could include such information for developers, users.
+Alternatively, another possibility would be to have a command which reveals the last error message with more detail: 
+`kdb last-message`.
+
+### Line 6: Module
+
+Reasons to keep it:
+* See the concrete plugin which causes the failure
+* Prohibits error masking (eg. `stat`ing a file happens in more plugins, devs will know where the error occurs)
```

Comment:
Not only devs: also users mount some plugin, which might cause unexpected errors later.